src/extension/wps/src/main/java/au/org/emii/wps/GoGoDuckProcess.java

```diff
@@ -63,87 +63,28 @@ public FileRawData execute(
             ProgressListener progressListener
     ) throws ProcessException {
         try {
-            final int threadCount = getThreadCount();
-            final int fileLimit = getFileLimit();
-
-            if (threadCount <= 0) {
+            if (goGoDuckConfig.getThreadCount() <= 0) {
                 throw new ProcessException("threadCount set to 0 or below, job will not run");
             }
 
-            if (fileLimit <= 0) {
-                throw new ProcessException("fileLimit set to 0 or below, job will not run");
-            }
-
             File outputFile = getResourceManager().getOutputResource(
                     getResourceManager().getExecutionId(true), layer + ".nc").file();
 
             String filePath = outputFile.toPath().toAbsolutePath().toString();
 
-            GoGoDuck ggd = new GoGoDuck(catalog, layer, subset, filePath, format, fileLimit);
+            GoGoDuck ggd = new GoGoDuck(catalog, layer, subset, filePath, format, goGoDuckConfig);
 
             ggd.setTmpDir(getWorkingDir());
-            ggd.setThreadCount(threadCount);
+            ggd.setThreadCount(goGoDuckConfig.getThreadCount());
```

Comment:
But we've already passed gogoduckconfig to gogoduck so we don't need to set the threadcount separately as well..